modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -593,97 +601,97 @@ public static boolean isMapSchema(Schema schema) {
      * @return true if the specified schema is an Array schema.
      */
     public static boolean isArraySchema(Schema schema) {
-        return (schema instanceof ArraySchema);
+        return schema instanceof ArraySchema || "array".equals(getSchemaType(schema));
```

Comment:
This check now allows 3.1 JsonSchema with type "array" which is the equivalent of the 3.0.X ArraySchema. This is the cause of the large number of changes around `ArraySchema` casts in this [commit](https://github.com/OpenAPITools/openapi-generator/pull/18034/commits/6ca1c41eecee89026861739aed951bb6f8661077).